tool/tctl/sso/configure/github.go

```diff
@@ -0,0 +1,145 @@
+
+	// ignores
+	ignoreMissingRoles := false
+	sub.Flag("ignore-missing-roles", "Ignore non-existing roles referenced in --claims-to-roles.").BoolVar(&ignoreMissingRoles)
+
+	sub.Alias("gh")
+
+	sub.Alias(`
+Examples:
+
+  > tctl sso configure gh -r octocats,admin,access,editor,auditor -r octocats,dev,access --secret GH_SECRET --id CLIENT_ID
+
+  Generate Github auth connector. Two role mappings are defined:
+    - members of 'admin' team in 'octocats' org will receive 'access', 'editor' and 'auditor' roles.
+    - members of 'dev' team in 'octocats' org will receive 'access' role.
+
+  The values for --secret and --id are provided by GitHub.
+
+  > tctl sso configure gh ... | tctl sso test
+  
+  Generate the configuration and immediately test it using "tctl sso test" command.`)
+
+	preset := &AuthKindCommand{
+		Run: func(clt auth.ClientI) error { return ghRunFunc(cmd, &spec, gh, clt) },
+	}
+
+	sub.Action(func(ctx *kingpin.ParseContext) error {
+		preset.Parsed = true
+		return nil
+	})
+
+	return preset
+}
+
+func ghRunFunc(cmd *SSOConfigureCommand, spec *types.GithubConnectorSpecV3, flags *ghExtraFlags, clt auth.ClientI) error {
+	allRoles, err := clt.GetRoles(context.TODO())
+	if err != nil {
+		cmd.Logger.WithError(err).Warn("unable to get roles list. Skipping attributes_to_roles sanity checks.")
+	} else {
+		roleMap := map[string]bool{}
+		var roleNames []string
+		for _, role := range allRoles {
+			roleMap[role.GetName()] = true
+			roleNames = append(roleNames, role.GetName())
+		}
+
+		for _, mapping := range spec.TeamsToLogins {
+			for _, role := range mapping.Logins {
+				_, found := roleMap[role]
+				if !found {
+					if flags.ignoreMissingRoles {
+						cmd.Logger.Warnf("teams-to-logins references non-existing role: %q. Available roles: %v.", role, roleNames)
+					} else {
+						return trace.BadParameter("teams-to-logins references non-existing role: %v. Correct the mapping, or add --ignore-missing-roles to ignore this error. Available roles: %v.", role, roleNames)
+					}
+				}
+			}
+		}
+	}
+
+	if spec.RedirectURL == "" {
+		cmd.Logger.Info("RedirectURL empty, resolving automatically.")
+		proxies, err := clt.GetProxies()
+		if err != nil {
+			cmd.Logger.WithError(err).Error("unable to get proxy list.")
+		}
+
+		// find first proxy with public addr
+		for _, proxy := range proxies {
+			publicAddr := proxy.GetPublicAddr()
+			if publicAddr != "" {
+				// TODO: double check this is correct
+				spec.RedirectURL = fmt.Sprintf("https://%v/v1/webapi/github/callback", publicAddr)
+				break
+			}
+		}
+
+		// check if successfully set.
+		if spec.RedirectURL == "" {
+			cmd.Logger.Warn("Unable to fill RedirectURL automatically, cluster's public address unknown.")
+		} else {
+			cmd.Logger.Infof("RedirectURL set to %q", spec.RedirectURL)
+		}
+	}
+
+	connector, err := types.NewGithubConnector(flags.connectorName, *spec)
+
+	if err != nil {
```

Comment:
nit: remove redundant space
```suggestion
	connector, err := types.NewGithubConnector(flags.connectorName, *spec)
	if err != nil {
```